packages/core/src/Utils/Analytics/analytics.ts

```diff
@@ -0,0 +1,46 @@
+interface AnalyticsEvent {
+    name: string;
+    properties: {
+        [key: string]: string;
+    };
+}
+
+let eventQueue: AnalyticsEvent[] = [];
+
+const setEvent = (event: AnalyticsEvent): void => {
+    eventQueue.push(event);
```

Comment:
you need to get the events from localstorage before push to it.
this way you always have the latest events stored in localstorage.